crates/bcr-ebill-persistence/src/db/nostr_contact_store.rs

```diff
@@ -65,6 +65,16 @@ impl NostrContactStoreApi for SurrealNostrContactStore {
         Ok(values.unwrap_or_default())
     }
 
+    /// Get all Nostr contacts from the store.
+    async fn get_all(&self) -> Result<Vec<NostrContact>> {
+        let result: Vec<NostrContactDb> = self.db.select_all(Self::TABLE).await?;
+        let values = result
+            .into_iter()
+            .map(|c| c.to_owned().try_into().ok())
+            .collect::<Option<Vec<NostrContact>>>();
+        Ok(values.unwrap_or_default())
```

Comment:
Hmm, doesn't this silently swallow the error, if e.g. all (or even 1) nostr contact in the DB would be invalid?

How about collecting into a `Result<Vec<NostrContact>>` and then propagating the error? I think even if this call would be "optional" in nature to succeed, this might help with debugging issues.